src/panoramic/cli/connections.py

```diff
@@ -0,0 +1,281 @@
+    password,
+    password_stdin,
+    database,
+    schema,
+    warehouse,
+    account,
+    project,
+    key_file,
+    no_test,
+):
+    """CLI command. Create new connection."""
+    connections = Connections.load()
+    if name in connections:
+        raise click.ClickException(f'Connection with name "{name}" already exists.')
+
+    if password_stdin:
+        password = click.prompt('Enter password: ', hide_input=True, type=str)
+
+    new_connection = {
+        'type': type,
+        'user': user,
+        'host': host,
+        'port': port,
+        'password': password,
+        'database': database,
+        'schema': schema,
+        'warehouse': warehouse,
+        'account': account,
+        'project': project,
+        'key_file': key_file,
+    }
+
+    credentials, error = get_dialect_credentials(new_connection)
+    if error != '':
+        raise click.ClickException(f'Failed to create connection: {error}')
+
+    if not no_test:
+        ok, error = Connections.test(credentials)
+        if not ok:
+            raise click.ClickException(f'Failed to create connection: {error}')
+
+    # Empty values are important for DBT credentials verification, but we don't need to store empty values in config.
+    connections[name] = {}
+    for key, value in new_connection.items():
+        if value:
+            connections[name][key] = value
+
+    Connections.save(connections)
+    echo_info('Connection was successfully created!')
+
+
+def update_connection_command(
+    name,
+    type,
+    user,
+    host,
+    port,
+    password,
+    password_stdin,
+    database,
+    schema,
+    warehouse,
+    account,
+    project,
+    key_file,
+    no_test,
+):
+    """CLI command. Update specific connection."""
+    connections = Connections.load()
+    if name not in connections:
+        raise click.ClickException(f'Connection with name "{name}" not found.')
+
+    if password_stdin:
+        password = click.prompt('Enter password: ', hide_input=True, type=str)
+
+    new_connection = connections[name].copy()
+    if type:
+        new_connection['type'] = type
+    if user:
+        new_connection['user'] = user
+    if password:
+        new_connection['password'] = password
+    if host:
+        new_connection['host'] = host
+    if port:
+        new_connection['port'] = port
+    if database:
+        new_connection['database'] = database
+    if schema:
+        new_connection['schema'] = schema
+    if warehouse:
+        new_connection['warehouse'] = warehouse
+    if account:
+        new_connection['account'] = account
+    if project:
+        new_connection['project'] = project
+    if key_file:
+        new_connection['key_file'] = key_file
+
+    # Fill with default values because DBT requires some fields we don't.
+    for key in CONNECTION_KEYS:
+        if key not in new_connection:
+            new_connection[key] = ''
+
+    credentials, error = get_dialect_credentials(new_connection)
+    if error != '':
```

Comment:
I don't mind using this "Go-flavoured" error handling but maybe it'd be nicer if we used `None` instead of a blank string `''`. Thoughts?

Reply:
Replaced for None.